From f0ee8d85e649106fdcc285b117b096e07d11fc00 Mon Sep 17 00:00:00 2001
From: Joe Williams <jwillia@iastate.edu>
Date: Mon, 11 May 2015 13:57:44 -0500
Subject: [PATCH] RESKC-395:Added the ability to view budget summary and print
 budget forms when proposal is view only

As an Approver and an Aggregator, I need to have access to the Budget Printed Reports for all status of proposals.
Issue: in the 1504 kuali co release; the Budget > Actions menu (where the Print menu for budget reports is located) does not appear to the user when the proposal is submitted to routing, and stays hidden when the proposal is submitted.
For non-s2s proposals, this leaves no method to access budget reports.

Scenario:
Create a proposal with a detailed budget that is going to an industrial or foundation sponsor.
Complete the required information to route the proposal.
Submit the proposal to routing.
As an APPROVER: attempt to print the Budget Summary Report from the Budget > Actions menu. There is no Action menu.
---
 .../impl/budget/core/ProposalBudgetPage.xml        | 30 ++++++++++++----------
 .../impl/budget/core/ProposalBudgetPrintForms.xml  |  6 +++--
 .../impl/budget/core/ProposalBudgetShared.xml      | 18 ++++++-------
 3 files changed, 30 insertions(+), 24 deletions(-)

diff --git a/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPage.xml b/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPage.xml
index f25b8a0..eb1aa13 100644
--- a/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPage.xml
+++ b/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPage.xml
@@ -47,19 +47,23 @@
                 <entry key="@{#line.budgetId == document.developmentProposal.finalBudget?.budgetId}" value="finalBudget"/>
             </map>
         </property>
-		<property name="lineActions">
-			<list>
-				<bean parent="PropDev-BudgetVersions-GenericActions" p:order="10">
-					<property name="listPrototype.items">
-						<list merge="true">
-							<bean parent="Uif-MenuDivider" p:order="40"/>
-			        		<bean parent="Uif-ActionLink" p:actionLabel="Include for Submission"
-								p:methodToCall="markForSubmission" p:additionalSubmitData="budgetId:@{#line.budgetId}"
-								p:iconClass="icon-ok" p:render="@{#line.budgetId != document.developmentProposal.finalBudget?.budgetId and canEditView}" p:order="50"/>
-                            <bean parent="Uif-ActionLink" p:actionLabel="Remove from Submission"
-                                  p:methodToCall="removeFromSubmission" p:iconClass="icon-ok"
-                                  p:render="@{#line.budgetId == document.developmentProposal.finalBudget?.budgetId and canEditView}" p:order="60"/>
-						</list>
+		<property name="items">
+			<list merge="true">
+				<bean parent="Uif-FieldGroupBase" p:label="Actions" p:order="100">
+					<property name="group">
+						<bean parent="PropDev-BudgetVersions-GenericActions">
+							<property name="listPrototype.items">
+								<list merge="true">
+									<bean parent="Uif-MenuDivider" p:order="40"/>
+									<bean parent="Uif-ActionLink" p:actionLabel="Include for Submission"
+										  p:methodToCall="markForSubmission" p:additionalSubmitData="budgetId:@{#line.budgetId}"
+										  p:iconClass="icon-ok" p:render="@{#line.budgetId != document.developmentProposal.finalBudget?.budgetId and canEditView}" p:order="50"/>
+									<bean parent="Uif-ActionLink" p:actionLabel="Remove from Submission"
+										  p:methodToCall="removeFromSubmission" p:iconClass="icon-ok"
+										  p:render="@{#line.budgetId == document.developmentProposal.finalBudget?.budgetId and canEditView}" p:order="60"/>
+								</list>
+							</property>
+						</bean>
 					</property>
 				</bean>
 			</list>
diff --git a/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPrintForms.xml b/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPrintForms.xml
index 17f7713..07b7cfb 100644
--- a/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPrintForms.xml
+++ b/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetPrintForms.xml
@@ -64,7 +64,8 @@
 		<property name="items">
 			<list>
 				<bean parent="Uif-DataField" p:propertyName="budgetReportName" p:label="Print Forms" p:order="10" />
-				<bean parent="Uif-InputField" p:propertyName="selectToComment" p:fieldLabel.labelText="[0]" p:order="20">
+				<bean parent="Uif-InputField" p:propertyName="selectToComment" p:fieldLabel.labelText="[0]" p:order="20"
+					  p:readOnly="false">
 					<property name="control">
 						<bean parent="Uif-CheckboxControl" />
 					</property>
@@ -84,7 +85,8 @@
                         </list>
                     </property>
 				</bean>
-				<bean parent="Uif-InputField" p:propertyName="selectToPrint" p:fieldLabel.labelText="[0]" p:order="30">
+				<bean parent="Uif-InputField" p:propertyName="selectToPrint" p:fieldLabel.labelText="[0]" p:order="30"
+					  p:readOnly="false">
 					<property name="control">
 						<bean parent="Uif-CheckboxControl" />
 					</property>
diff --git a/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetShared.xml b/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetShared.xml
index 00c7d8f..8638d84 100644
--- a/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetShared.xml
+++ b/coeus-impl/src/main/resources/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetShared.xml
@@ -28,7 +28,7 @@
 		parent="Uif-TableCollectionSection" p:title="Budget"
 		p:headerText="Budgets" 
 		p:collectionObjectClass="org.kuali.coeus.propdev.impl.budget.ProposalDevelopmentBudgetExt"
-		p:renderAddLine="false" p:renderLineActions="true" p:layoutManager.renderSequenceField="false">
+		p:renderAddLine="false" p:renderLineActions="false" p:layoutManager.renderSequenceField="false">
 		<property name="items">
 			<list>
 				<bean parent="Uif-SimpleFieldGroup" p:order="10" p:label="Name">
@@ -77,11 +77,11 @@
 				<bean parent="Uif-DataField" p:label="End" p:propertyName="summaryPeriodEndDate" p:order="70" />
 				<bean parent="Uif-DataField" p:label="Status" p:propertyName="budgetStatusDo.description" p:order="80"/>
 				<bean parent="Uif-DataField" p:propertyName="comments" p:order="90" />
-			</list>
-		</property>
-		<property name="lineActions">
-			<list>
-				<bean parent="PropDev-BudgetVersions-GenericActions" p:order="10"/>
+				<bean parent="Uif-FieldGroupBase" p:label="Actions" p:order="100">
+					<property name="group">
+						<bean parent="PropDev-BudgetVersions-GenericActions" p:order="10"/>
+					</property>
+				</bean>
 			</list>
 		</property>
 		<property name="header.rightGroup">
@@ -121,13 +121,13 @@
 		<property name="listPrototype.items">
 			<list>
 				<bean parent="Uif-ActionLink" p:actionLabel="View Summary" p:methodToCall="populateBudgetSummary" p:additionalSubmitData="budgetId:@{#line.budgetId}" p:iconClass="icon-eye-open" p:order="10"/>
-				<bean parent="Uif-ActionLink" p:actionLabel="Copy" p:iconClass="icon-copy" p:render="@{#view.editModes.containsKey(#budgetAuthConsts.ADD_BUDGET_EDIT_MODE)}"
+				<bean parent="Uif-ActionLink" p:actionLabel="Copy" p:iconClass="icon-copy" p:render="@{#view.editModes.containsKey(#budgetAuthConsts.ADD_BUDGET_EDIT_MODE) and !viewOnly and canEditView}"
 					p:actionScript="showDialog('PropDev-BudgetPage-CopyBudgetDialog', {showHandler: function(e) {Kc.PropDev.Budget.copyBudget(@{#line.budgetId}, e)}})"
 					p:order="20"/>
 				<bean parent="Uif-ActionLink" p:actionLabel="Print" p:methodToCall="populatePrintForms" p:additionalSubmitData="budgetId:@{#line.budgetId}" p:iconClass="icon-print" p:order="30" />
                 <bean parent="Uif-ActionLink" p:actionLabel="Complete Budget" p:methodToCall="markBudgetVersionComplete"
                       p:additionalSubmitData="budgetId:@{#line.budgetId}" p:iconClass="icon-check"
-                      p:render="@{#lp.budgetStatusDo.description != 'Complete'}" p:order="35">
+                      p:render="@{#lp.budgetStatusDo.description != 'Complete' and !viewOnly and canEditView}" p:order="35">
                     <property name="confirmationDialog">
                         <bean id="PropDev-BudgetVersions-CompleteBudgetConfirm" parent="Uif-DialogGroup-OkCancel"
                               p:promptText="Are you sure you want to set this budget to complete?" />
@@ -135,7 +135,7 @@
                 </bean>
                 <bean parent="Uif-ActionLink" p:actionLabel="Mark Budget Incomplete" p:methodToCall="markBudgetVersionIncomplete"
                       p:additionalSubmitData="budgetId:@{#line.budgetId}" p:iconClass="icon-close"
-                      p:render="@{#lp.budgetStatusDo.description == 'Complete'}" p:order="37">
+                      p:render="@{#lp.budgetStatusDo.description == 'Complete' and !viewOnly and canEditView}" p:order="37">
                     <property name="confirmationDialog">
                         <bean id="PropDev-BudgetVersions-InCompleteBudgetConfirm" parent="Uif-DialogGroup-OkCancel"
                               p:promptText="Are you sure you want to set this budget to incomplete?" />
